base/reduce.jl

```diff
@@ -579,11 +569,12 @@ true
 ```
 """
 any(f::Any, itr) = any(Predicate(f), itr)
-any(f::Predicate, itr) = mapreduce_sc_impl(f, |, itr)
-any(f::typeof(identity), itr) =
-    eltype(itr) <: Bool ?
-        mapreduce_sc_impl(f, |, itr) :
-        reduce(or_bool_only, false, itr)
+function any(f::Predicate, itr)
```

Comment:
It looks as though if I change `mapreduce` to do that directly, there's no need for any of the other `mapreduce_*sc*` functions.